sqlite.go

```diff
@@ -363,8 +385,31 @@ func (r *rows) Next(dest []driver.Value) (err error) {
 				if err != nil {
 					return err
 				}
-
+				cty, err := r.columnDeclType(i)
```

Comment:
> Atleast if we want to handle int->boolean conversion by default, which seems like a straightforward thing to do?

Whatever it makes compatible in compatibility mode. No guesswork please in default mode because [sqlite3_column_decltype](https://sqlite.org/c3ref/column_decltype.html) is unreliable:

> If the Nth column of the result set is an expression or subquery, then a NULL pointer is returned.

That means going from, for example, `boolColumn` to `boolColumn or otherBoolColumn` in a query result in suddenly changed type of the value and Scan will fail. Analogically for time/date stuff.